connect/mirror/src/main/java/org/apache/kafka/connect/mirror/OffsetSyncStore.java

```diff
@@ -76,26 +112,26 @@ OptionalLong translateDownstream(TopicPartition sourceTopicPartition, long upstr
     }
 
     // poll and handle records
-    synchronized void update(Duration pollTimeout) {
+    synchronized void update(Duration pollTimeout) throws TimeoutException {
         try {
-            consumer.poll(pollTimeout).forEach(this::handleRecord);
-        } catch (WakeupException e) {
+            backingStore.readToEnd().get(pollTimeout.toMillis(), TimeUnit.MILLISECONDS);
+        } catch (WakeupException | InterruptedException | ExecutionException e) {
             // swallow
```

Comment:
This is much cleaner, as it puts the responsibility for balking on the OffsetSyncStore, rather than sharing the responsibility with the CheckpointTask. Thanks for the suggestion!